src/asymmetric_crypto_ctx.cpp

```diff
@@ -284,17 +282,43 @@ namespace mococrw {
      * Implements the retrieving of the hash function to be used of the stored padding object
      */
     template <class Key>
-    class RSASignatureImpl : public RSAImpl<Key, RSASignaturePadding>, public SignatureCtxImpl {
+    class RSASignatureImpl : public RSAImpl<Key, RSASignaturePadding> {
     public:
         RSASignatureImpl(const Key& key, openssl::DigestTypes hashFunction, std::shared_ptr<RSASignaturePadding> padding)
-            : RSAImpl<Key, RSASignaturePadding>(key, padding), SignatureCtxImpl(hashFunction) {}
+            : RSAImpl<Key, RSASignaturePadding>(key, padding), hashFunction(hashFunction) {}
+    protected:
+        openssl::DigestTypes hashFunction;
```

Comment:
This library prefixes class members with _

Reply:
Done